libraries/abstractions/ble_hal/test/src/iot_test_ble_hal_afqp.c

```diff
@@ -665,8 +685,11 @@ TEST( Full_BLE, BLE_Property_ReadLongCharacteristic )
     BTGattResponse_t xGattResponse;
     BLETESTconfirmCallback_t xConfirmEvent;
     BTStatus_t xStatus;
+    uint16_t usPayloadLength;
+    uint16_t usOffset = 0;
 
     memset( LongReadBuffer, 49, bletests_LONG_READ_LEN * sizeof( uint8_t ) );
+    usPayloadLength = _bletestsMTU_SIZE > 512 ? 512 : _bletestsMTU_SIZE - 1;
```

Comment:
Why "49" above, and "512" on this line?

Reply:
the data that RPi send is "1111111...", 49 is the ascii code of '1'. 512 is the longest attribute length that we can send.